q2_diversity/_beta/_visualizer.py

```diff
@@ -222,3 +223,71 @@ def beta_group_significance(output_dir: str,
         'result': result_html,
         'pairwise_results': pairwise_results_html
     })
+
+
+def _metadata_distance(metadata: pd.Series)-> skbio.DistanceMatrix:
+    # This code is derived from @jairideout's scikit-bio cookbook recipe,
+    # "Exploring Microbial Community Diversity"
+    # https://github.com/biocore/scikit-bio-cookbook
+    distances = scipy.spatial.distance.pdist(
+        metadata.values[:, numpy.newaxis], metric='euclidean')
+    return skbio.DistanceMatrix(distances, ids=metadata.index)
+
+
+def beta_correlation(output_dir: str,
+                     distance_matrix: skbio.DistanceMatrix,
+                     metadata: qiime2.MetadataCategory,
+                     method: str='spearman',
+                     permutations: int=999) -> None:
+    try:
+        metadata = pd.to_numeric(metadata.to_series(), errors='raise')
+    except ValueError as e:
+        raise ValueError('Only numeric data can be used with the Mantel test. '
+                         'Non-numeric data was encountered in the sample '
+                         'metadata. Orignal error message follows:\n%s' %
+                         str(e))
+
+    initial_metadata_length = len(metadata)
+    metadata = metadata.loc[list(distance_matrix.ids)]
+    metadata = metadata.replace(r'', numpy.nan).dropna()
+    filtered_metadata_length = len(metadata)
+
+    ids_with_missing_metadata = set(distance_matrix.ids) - set(metadata.index)
+    if len(ids_with_missing_metadata) > 0:
+        raise ValueError('All samples in distance matrix must be present '
+                         'and contain data in the sample metadata. The '
+                         'following samples were present in the distance '
+                         'matrix, but were missing from the sample metadata '
+                         'or had no data: %s' %
+                         ', '.join(ids_with_missing_metadata))
+
+    metadata_distances = _metadata_distance(metadata)
+    r, p, n = skbio.stats.distance.mantel(
+        distance_matrix, metadata_distances, method, permutations)
+
+    result = pd.Series([method.title(), n, permutations, r, p],
+                       index=['Method', 'Sample size', 'Permutations',
+                              'r statistic', 'p-value'],
+                       name='Mantel test results')
```

Comment:
Can you include the metadata category name somewhere on the results page? It might work in this header, but if you have another place where it'd be easily noticeable let's go with that. I realize the category name is on the y-axis of the plot but it'd be good to have it in text form somewhere at the top of the page.